Решение по 14 спринту

iliaalii opened the pull request:
- добавлены контроллеры item и user
 - добавлены хранилища для item и user
 - Добавлены сервисы
 - Добавлены dto и мапперы

iliaalii committed:
настройка начального каркаса

iliaalii committed:
Работа по 14 спринту:
 - добавлены контроллеры item и user
 - добавлены хранилища для item и user
 - Добавлены сервисы
 - Добавлены dto и мапперы

avfyodorov commented on src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java:
```diff
@@ -0,0 +1,43 @@
+import ru.practicum.shareit.item.dto.ItemDto;
+import ru.practicum.shareit.item.model.Item;
+
+@Component
```
Аннотация @Component не требуется, так как класс не имеет состояния(все static), а значит создание его объекта не имеет никакого смысла, так как все принадлежит классу.  Для информации. 
Чтобы создать приватный конструктор без параметров, тем самым запретив создание объектов этого класса,
можно добавить такую аннотацию @NoArgsConstructor(access = AccessLevel.PRIVATE) или @UtilityClass

iliaalii replied:
Принял, исправил. В данно случае решил выбрать второй вариант

avfyodorov commented on src/main/java/ru/practicum/shareit/user/mapper/UserMapper.java:
```diff
@@ -0,0 +1,41 @@
+import ru.practicum.shareit.user.dto.UserDto;
+import ru.practicum.shareit.user.model.User;
+
+@Component
```
аналогично...можно добавить такую аннотацию @NoArgsConstructor(access = AccessLevel.PRIVATE) или @UtilityClass

avfyodorov commented on src/main/java/ru/practicum/shareit/item/model/Item.java:
```diff
@@ -1,7 +1,15 @@
+    String description;
+    boolean available;
+    Integer owner;
+    Integer request;
```
Для соблюдения инкапсуляции все поля лучше сделать приватными.

iliaalii replied:
надо выработать привычку делать сразу -_- 
исправил

avfyodorov commented on src/main/java/ru/practicum/shareit/user/model/User.java:
```diff
@@ -0,0 +1,12 @@
+public class User {
+    Integer id;
+    String name;
+    String email;
```
Для соблюдения инкапсуляции все поля лучше сделать приватными.

avfyodorov reviewed (commented):
Добрый вечер, Илья! 

Хорошая работа, всего несколько небольших уточнений.

iliaalii committed:
Исправления по 14 спринту:
 - поля item и user, стали приватными
 - в мапперах изменена аннотация на @UtilityClass, вместо @Component

avfyodorov reviewed (commented):
Добрый вечер, Илья!

Замечаний нет.
Работа принята.
